esmvalcore/preprocessor/_io.py

```diff
@@ -269,3 +280,63 @@ def _write_ncl_metadata(output_dir, metadata):
     write_ncl_settings(info, filename)
 
     return filename
+
+
+def _concatenate_two_overlapping_cubes(cubes):
+    """Concatenate time-overlapping cubes (two cubes only)."""
+    # we arrange [cube1, cube2] so that cube1.start <= cube2.start
+    if cubes[0].coord('time').points[0] <= cubes[1].coord('time').points[0]:
+        cubes = [cubes[0], cubes[1]]
+    else:
+        cubes = [cubes[1], cubes[0]]
+
+    # get time end points
+    time_1 = cubes[0].coord('time')
+    time_2 = cubes[1].coord('time')
+    data_start_1 = time_1.cell(0).point
+    data_start_2 = time_2.cell(0).point
+    data_end_1 = time_1.cell(-1).point
+    data_end_2 = time_2.cell(-1).point
+
+    # case 1: both cubes start at the same time -> return longer cube
+    if data_start_1 == data_start_2:
+        if data_end_1 < data_end_2:
+            cubes = [cubes[1]]
+        else:
+            cubes = [cubes[0]]
```

Comment:
@valeriupredoi This case is not covered in the tests yet.

Reply:
good call, mate! Now it has full coverage :beer: